pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -255,16 +261,54 @@ func Convert_v1_Disk_To_api_Disk(c *ConverterContext, diskDevice *v1.Disk, disk
 	return nil
 }
 
-func checkDirectIOFlag(path string) bool {
-	// check if fs where disk.img file is located or block device
-	// support direct i/o
+type DirectIOChecker interface {
+	Check(path string, isBlockDev bool) (bool, error)
+}
+
+type directIOChecker struct{}
+
+func NewDirectIOChecker() DirectIOChecker {
+	return &directIOChecker{}
+}
+
+// based on https://gitlab.com/qemu-project/qemu/-/blob/master/util/osdep.c#L344
+func (c *directIOChecker) Check(path string, isBlockDev bool) (bool, error) {
```

Comment:
I have an issue with the boolean arg: it exposes that this function has two quite different functionalities - one for files and the other for block devices. As a reader I'd prefer `blockDevSupportsDirectIO(path)` over `DirectIOChecker.check(path, false`). I think that this split would allow the implementation of `fileSupportsDirectIO` to be simpler, always creating `TempFile(Dir(path))` and attempting to read it with `O_DIRECT`

Reply:
Right, 'single responsibility' principle :) Agree, makes sense to have two separate methods